string alg
string descriptor
string rate
---
string acknowledgment
